soda/core/soda/execution/check/profile_columns_run.py

```diff
@@ -46,96 +47,88 @@ def __init__(self, data_source_scan: DataSourceScan, profile_columns_cfg: Profil
         self.profile_columns_cfg: ProfileColumnsCfg = profile_columns_cfg
         self.logs = self.data_source_scan.scan._logs
 
+    @staticmethod
+    def parse_profiling_expressions(profiling_expressions: list[str]) -> list[list[str]]:
+        parsed_profiling_expressions = []
+        for profiling_expression in profiling_expressions:
+            table_name, column_name = profiling_expression.split(".")
+            table_name_operator = "LIKE" if "%" in table_name else "="
+            column_name_operator = "LIKE" if "%" in column_name else "="
+
+            parsed_profiling_expressions.append([table_name, table_name_operator, column_name, column_name_operator])
+        return parsed_profiling_expressions
+
     def run(self) -> ProfileColumnsResult:
         profile_columns_result: ProfileColumnsResult = ProfileColumnsResult(self.profile_columns_cfg)
         self.logs.info(f"Running column profiling for data source: {self.data_source.data_source_name}")
-        include_tables = self._get_table_expression(self.profile_columns_cfg.include_columns)
-        exclude_tables = self._get_table_expression(self.profile_columns_cfg.exclude_columns)
-
-        # row_counts is a dict that maps table names to row counts.
-        table_names: dict[str, int] = self.data_source.get_table_names(
-            include_tables=include_tables,
-            exclude_tables=exclude_tables,
-            query_name="profile-columns-get-table-names",
+
+        include_patterns = self.parse_profiling_expressions(self.profile_columns_cfg.include_columns)
+        exclude_patterns = self.parse_profiling_expressions(self.profile_columns_cfg.exclude_columns)
+
+        tables_columns_metadata: defaultdict[str, dict] = self.data_source.get_tables_columns_profiling(
+            include_patterns=include_patterns,
+            exclude_patterns=exclude_patterns,
+            query_name=f"profile-columns-get-table-and-column-metadata",
         )
 
-        if len(table_names) < 1:
+        if tables_columns_metadata is None:
             self.logs.warning(
-                f"No table matching your SodaCL inclusion list found on your {self.data_source.data_source_name} "
-                "data source. Profiling results may be incomplete or entirely skipped",
+                f"Your SodaCL profiling expressions did not return any existing dataset name + column name combinations for your '{self.data_source.data_source_name}' "
+                f"data source. Please make sure that the patterns in your profiling expressions define existing dataset name + column name combinations."
+                f" Profiling results may be incomplete or entirely skipped. See the docs for more information: \n"
+                f"https://docs.soda.io/soda-cl/profile.html#display-profile-information-in-soda-cloud",
```

Comment:
ask maybe @janet-can to create a dynamic link for the doc.

Reply:
Of course!  Here you go: https://go.soda.io/display-profile